vignettes/advanced-use.Rmd

```diff
@@ -0,0 +1,58 @@
+---
+title: "Advanced Uses"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{advanced-use}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup}
+library(npi)
+library(purrr)
+```
+
+This vignette provides an brief exploration to advanced uses of the npi package.
+
+## Run `npi_search()` on multiple search terms
+
+Users can serially execute searches for multiple values of a single query parameter by using `npi` in combination with the [`purrr`](https://purrr.tidyverse.org/) package. In the example below, we search multiple NPI numbers. A single tibble is returned with record information corresponding toe matching records.
```

Comment:
It would also be nice to show an example using base R with a simple `for` loop alongside the purrr version here. I'm a fan of the tidyverse, but not everyone uses it, and arguably a simple `for` loop better expresses the intent of the code to people less familiar with functional programming style. I won't reject the PR if this isn't added, but I wanted to capture the idea inline here.

Reply:
Done.